# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1324485731 28800
#      Wed Dec 21 08:42:11 2011 -0800
# Branch stable
# Node ID 8717cc808b2008fddb2afad1fe1f4a8f7b172753
# Parent  97f41a44dd808a59859327efed5678c51c58a162
Stop memory leak in idx_vector_rep when constructor fails.

* idx-vector.cc:  Stop memory leak in idx_vector_rep when constructor fails.

diff --git a/liboctave/idx-vector.cc b/liboctave/idx-vector.cc
--- a/liboctave/idx-vector.cc
+++ b/liboctave/idx-vector.cc
@@ -326,17 +326,20 @@ idx_vector::idx_vector_rep::idx_vector_r
   if (len != 0)
     {
       octave_idx_type *d = new octave_idx_type[len];
       for (octave_idx_type i = 0; i < len; i++)
         d[i] = convert_index (nda.xelem (i), err, ext);
       data = d;
 
       if (err)
+      {
+        delete [] data;
         gripe_invalid_index ();
+      }
     }
 }
 
 // Note that this makes a shallow copy of the index array.
 
 idx_vector::idx_vector_rep::idx_vector_rep (const Array<octave_idx_type>& inda)
   : data (inda.data ()), len (inda.numel ()), ext (0),
     aowner (new Array<octave_idx_type> (inda)), orig_dims (inda.dims ())
